src/planner/binder/expression/bind_operator_expression.cpp

```diff
@@ -136,6 +137,17 @@ BindResult ExpressionBinder::BindExpression(OperatorExpression &op, idx_t depth)
 		D_ASSERT(op.children[i]->expression_class == ExpressionClass::BOUND_EXPRESSION);
 		children.push_back(std::move(BoundExpression::GetExpression(*op.children[i])));
 	}
+
+	if (op.type == ExpressionType::COMPARE_IN || op.type == ExpressionType::COMPARE_NOT_IN) {
+		auto result_type = children[0]->return_type;
+		for (idx_t i = 1; i < children.size(); ++i) {
+			result_type = BoundComparisonExpression::BindComparison(result_type, children[i]->return_type);
+		}
+		for (idx_t i = 0; i < children.size(); ++i) {
+			PushCollation(context, children[i], result_type, true);
```

Comment:
Can this not be moved down below the `ResolveOperatorType` which should have already done the resolution of `result_type`?

Reply:
I think you are right. Thanks for your carefully reviewing!